readthedocs/projects/static-src/projects/js/import.js

```diff
@@ -0,0 +1,252 @@
+        }
+    });
+    self.private = ko.observable(instance.private);
+    self.active = ko.observable(instance.active);
+    self.admin = ko.observable(instance.admin);
+    self.is_locked = ko.computed(function () {
+        return (self.private() && !self.admin());
+    });
+    self.avatar_url = ko.observable(
+        append_url_params(instance.avatar_url, {size: 32})
+    );
+
+    self.import_repo = function () {
+        var data = {
+                name: self.name(),
+                repo: self.clone_url(),
+                repo_type: self.vcs(),
+                description: self.description(),
+                project_url: self.html_url(),
+            },
+            form = $('<form />');
+
+        form
+            .attr('action', view.urls.projects_import)
+            .attr('method', 'POST');
+
+        Object.keys(data).map(function (attr) {
+            var field = $('<input>')
+                .attr('type', 'hidden')
+                .attr('name', attr)
+                .attr('value', data[attr]);
+            form.append(field);
+        });
+
+        var token = csrf.get_cookie('csrftoken'),
+            csrf_field = $('<input>');
+
+        csrf_field
+            .attr('type', 'hidden')
+            .attr('name', 'csrfmiddlewaretoken')
+            .attr('value', token);
+        form.append(csrf_field);
+
+        form.submit();
+    };
+}
+
+function ProjectImportView (instance, urls) {
+    var self = this,
+        instance = instance || {};
+
+    self.urls = urls || {};
+
+    // For task display
+    self.error = ko.observable(null);
+    self.is_syncing = ko.observable(false);
+    self.is_ready = ko.observable(false);
+
+    // For filtering
+    self.page_count = ko.observable(null);
+    self.page_current = ko.observable(null);
+    self.page_next = ko.observable(null);
+    self.page_previous = ko.observable(null);
+    self.filter_org = ko.observable(null);
+
+    self.organizations_raw = ko.observableArray();
+    self.organizations = ko.computed(function () {
+        var organizations = [],
+            organizations_raw = self.organizations_raw();
+        for (n in organizations_raw) {
+            var organization = new Organization(organizations_raw[n], self);
+            organizations.push(organization);
+        }
+        return organizations;
+    });
+    self.projects = ko.observableArray();
+
+    ko.computed(function () {
+        var org = self.filter_org(),
+            orgs = self.organizations(),
+            url = self.page_current() || self.urls['remoterepository-list'];
+
+        if (org) {
+            url = append_url_params(url, {org: org});
+        }
+
+        $.getJSON(url)
+            .success(function (projects_list) {
+                var projects = [];
+                self.page_next(projects_list.next);
+                self.page_previous(projects_list.previous);
+
+                for (n in projects_list.results) {
+                    // TODO replace org id here
```

Comment:
Oops, actually this should be cleaned up, this was from a previous attempt at associating projects <-> organization, which isn't needed as this is an API feature now.